### ISSUE TITLE
Wrong warning message when defining the property quarkus.rest.path

### ISSUE THREAD

**Issue by iompo** [quarkusio/quarkus#39859]
### Describe the bug

When compiling the app with the module `quarkus-rest` installed and the property `quarkus.rest.path` defined, then the following log is printed:
```
Unrecognized configuration key "quarkus.rest.path" was provided; it will be ignored; verify that the dependency extension for this configuration is set or that you did not make a typo
```
It shouldn't since the property exists and is correctly used at runtime.

### Expected behavior

No warning message in the log

### Actual behavior

The following log message is printed:
```
Unrecognized configuration key "quarkus.rest.path" was provided; it will be ignored; verify that the dependency extension for this configuration is set or that you did not make a typo
```

### How to Reproduce?

Define the property `quarkus.rest.path` within the application.properties file and then compile the app with the quarkus-rest module installed.

### Quarkus version or git rev

3.9.1

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

gradle 8.6

**Comment by gsmet** [comment]
Yeah, it's going to get fixed in 3.9.2 that I'm releasing as we speak.
